src/main/java/com/avioconsulting/mule/logger/internal/CustomLoggerOperation.java

```diff
@@ -2,31 +2,132 @@
 
 import static org.mule.runtime.extension.api.annotation.param.MediaType.ANY;
 
-import com.avioconsulting.mule.logger.api.processor.AdditionalProperties;
-import com.avioconsulting.mule.logger.api.processor.ExceptionProperties;
-import com.avioconsulting.mule.logger.api.processor.LogProperties;
-import com.avioconsulting.mule.logger.api.processor.MessageAttributes;
+import com.avioconsulting.mule.logger.api.processor.*;
 import com.avioconsulting.mule.logger.internal.config.CustomLoggerConfiguration;
+import com.mulesoft.modules.cryptography.api.jce.config.JceEncryptionPbeAlgorithm;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import javax.inject.Inject;
+import org.mule.extension.compression.api.strategy.CompressorStrategy;
+import org.mule.extension.compression.api.strategy.gzip.GzipCompressorStrategy;
 import org.mule.runtime.api.component.location.ComponentLocation;
+import org.mule.runtime.api.streaming.CursorProvider;
+import org.mule.runtime.api.streaming.bytes.CursorStream;
+import org.mule.runtime.core.api.util.Base64;
+import org.mule.runtime.core.api.util.IOUtils;
 import org.mule.runtime.extension.api.annotation.param.Config;
 import org.mule.runtime.extension.api.annotation.param.MediaType;
 import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+import org.mule.runtime.extension.api.client.DefaultOperationParameters;
+import org.mule.runtime.extension.api.client.DefaultOperationParametersBuilder;
+import org.mule.runtime.extension.api.client.ExtensionsClient;
+import org.mule.runtime.extension.api.runtime.operation.Result;
 import org.mule.runtime.extension.api.runtime.parameter.CorrelationInfo;
+import org.mule.runtime.extension.api.runtime.parameter.ParameterResolver;
+import org.mule.runtime.extension.api.runtime.streaming.StreamingHelper;
 
 /**
  * Author: Chakri Bonthala
  */
 public class CustomLoggerOperation {
 
+  @Inject
+  ExtensionsClient extensionsClient;
+
   @MediaType(value = ANY, strict = false)
   public void log(@ParameterGroup(name = "Log") LogProperties logProperties,
       @ParameterGroup(name = "Message Attributes") MessageAttributes messageAttributes,
       @ParameterGroup(name = "Exception Details") ExceptionProperties exceptionProperties,
       @ParameterGroup(name = "Additional Options") AdditionalProperties additionalProperties,
       @Config CustomLoggerConfiguration loggerConfig,
       ComponentLocation location,
-      CorrelationInfo correlationInfo) {
+      CorrelationInfo correlationInfo,
+      StreamingHelper streamingHelper) {
+
+    transformPayload(logProperties, loggerConfig, streamingHelper);
     loggerConfig.getLogger().log(logProperties, messageAttributes, exceptionProperties, additionalProperties,
         loggerConfig, location, correlationInfo.getCorrelationId());
   }
+
```

Comment:
Could we add some javadoc comments to new methods?

Reply:
Added